.travis.yml

```diff
@@ -16,6 +16,7 @@ before_install:
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
   - npm install --quiet -g gulp
+  - cd demo && npm install && cd ..
```

Comment:
Indeed, it should work like this but no. An issue is open : https://github.com/angular/angular-cli/issues/3864 . But I agree with you, duplicate node-modules is not the ideal solution.